### PR TITLE
Add EIP-4844 presets + SpecConfig

### LINKED ISSUE #6379: Define new presets
### new presets:
- [ ] [mainnet](https://github.com/ethereum/consensus-specs/blob/dev/presets/mainnet/eip4844.yaml) 
- [ ] [minimal](https://github.com/ethereum/consensus-specs/blob/dev/presets/minimal/eip4844.yaml) 

new fields:
-`FIELD_ELEMENTS_PER_BLOB`
-`MAX_BLOBS_PER_BLOCK`

### new fork in configurations for [mainnet](https://github.com/ethereum/consensus-specs/blob/dev/configs/mainnet.yaml) and [minimal](https://github.com/ethereum/consensus-specs/blob/dev/configs/minimal.yaml) 
- [ ] `EIP4844_FORK_VERSION`
- [ ] `EIP4844_FORK_EPOCH`

### PR DESCRIPTION
**Pull request by zilm13**
## PR Description
Adds EIP-4844 presets to `mainnet`, `minimal`. Also added to `swift` as it's artificial config used for testing.
For other networks I've found configs for 
- Gnosis Testnet https://github.com/gnosischain/configs/blob/main/chiado/config.yaml
- Gnosis Main https://github.com/gnosischain/configs/blob/main/mainnet/config.yaml
- Prater https://github.com/eth-clients/goerli/blob/main/prater/config.yaml
- Sepolia https://github.com/eth-clients/sepolia/blob/main/bepolia/config.yaml

None of them contains EIP-4844 fork, so I have not added anything to them.
I also have not added anything to https://github.com/ConsenSys/teku/tree/master/ethereum/spec/src/test/resources/tech/pegasys/teku/spec/config/standard because it requires `SpecConfig` implementation to pass tests. It looks like the part of #6401 

## Fixed Issue(s)

Fixes #6379 

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

**Review by tbenr — APPROVED** [review]
LGTM

**Comment by zilm13** [comment]
I will merge it with #6401 PR, because there is no way to add vars before adding them to `SpecConfig`

**Comment by zilm13** [comment]
Implemented `SpecConfig`
Skipped `SpecMilestone` and `SpecFactory` (and some tests like `SpecConfigBuilderTest`) as they requires `SpecLogic` implementation

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/config/SpecConfigBuilder.java [hunk 1]
```diff
@@ -176,8 +180,13 @@ public SpecConfig build() {
         bellatrixConfig = Optional.of(bellatrixBuilder.get().build(altairConfig));
         config = bellatrixConfig.get();
       }
+      Optional<SpecConfigCapella> capellaConfig = Optional.empty();
       if (capellaBuilder.isPresent() && capellaBuilder.get().isCapellaIncluded()) {
-        config = capellaBuilder.get().build(bellatrixConfig.orElseThrow());
+        capellaConfig = Optional.of(capellaBuilder.get().build(bellatrixConfig.orElseThrow()));
```
Comment (tbenr) [inline ethereum/spec/src/main/java/tech/pegasys/teku/spec/config/SpecConfigBuilder.java]: Starting from capella we changed the pattern here. We were nesting the altair and bellatrix but then the other are all under altair. Side effect could be that we load config for a fork (ie Capella) without previous fork config being defined (ie Bellatrix).

**Review by tbenr — APPROVED** [review]
LGTM with a comment on `SpecConfig::build()`

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/config/SpecConfigBuilder.java (hunk 1 above)
Comment (zilm13) [reply ethereum/spec/src/main/java/tech/pegasys/teku/spec/config/SpecConfigBuilder.java]: How about this? https://github.com/ConsenSys/teku/pull/6402/commits/dd11f012ed056d891f728efb89fe27a108e8cef5

**Review by tbenr — APPROVED** [review]
LGTM i like the approach.
A nit is adding some `final` in local vars in `getActiveBuilders`

**Pull request merged**